src/ReflectionMagic/Property.cs

```diff
@@ -38,13 +38,16 @@ void IProperty.SetValue(object obj, object value, object[] index)
             else
             {
                 var backingFieldName = $"<{_propertyInfo.Name}>k__BackingField";
-                var type = obj.GetType();
-                var backingField = type.GetField(backingFieldName, BindingFlags.Instance | BindingFlags.NonPublic);
-                if (backingField == null)
+                for (var type = obj.GetType(); type != null; type = type.BaseType)
                 {
-                    throw new MissingMemberException($"The property {type}.{_propertyInfo.Name} does not have a setter nor a backing field ({backingFieldName}).");
+                    var backingField = type.GetField(backingFieldName, BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
And I assume there is no special flag like `FlattenHierarchy` that would do this automatically because it's private?

Reply:
I learnt about `FlattenHierarchy` yesterday so if was the first thing I tried! But it did not work so I had to loop over the type hierarchy instead.

Reply:
Yep, not surprising that it didn't.

Reply:
I'm surprised it didn't work initially as you usually have to explicitly filter out inherited properties using `DeclaredOnly`.